src/charm.py

```diff
@@ -86,14 +87,18 @@ def __init__(self, *args):
         # TODO:
         #  ingress route provisioning a separate TCP ingress for each receiver if GRPC doesn't work directly
 
-        self._ingress = IngressPerAppRequirer(
-            self, port=self.tempo.tempo_server_port, strip_prefix=True
-        )
-
+        self._ingress = TraefikRouteRequirer(self, self.model.get_relation("ingress"), "ingress")  # type: ignore
         self._tracing = TracingEndpointProvider(
-            self, host=self.tempo.host, external_url=self._ingress.url
+            self, host=self.tempo.host, external_url=self._ingress.external_host
         )
 
+        self.framework.observe(self.on["ingress"].relation_joined, self._configure_ingress)
+        self.framework.observe(self.on.leader_elected, self._configure_ingress)
+        self.framework.observe(self.on.config_changed, self._configure_ingress)
+        self.framework.observe(self._ingress.on.ready, self._on_ingress_ready)  # pyright: ignore
+        # TODO there's no revoked ingress action with traefik_route?
```

Comment:
no, you have to listen to broken/departed separately. Or contribute one to traefik-route I guess